eng/pipelines/templates/Apex_Tests_On_Windows.yml

```diff
@@ -85,8 +87,36 @@ stages:
 
               $(DartLab.Path)\Scripts\VisualStudio\Build\Get-BaselineBuildCommitIDs.ps1 -BuildJson $baselineBuildsFile -OutVariableName "BaselineBuildCommitIds"
               } catch {
-                Write-Host "##vso[task.LogIssue type=error;]Unable to get Baseline build commit ids: $_"
-              }            
+                Write-Host "Unable to get Baseline build commit ids: $_"
+              }
+        - task: PowerShell@2
+          name: SetVisualStudioBaseBuildID
+          displayName: Set 'VisualStudio.BaseBuild.ID'
+          retryCountOnTaskFailure: 3
+          condition: ne(variables['BaselineBuildCommitIds'], '')
+          continueOnError: true
+          inputs:
+            filePath: $(DartLab.Path)\Scripts\VisualStudio\Build\Get-BaselineBuildID.ps1
+            arguments: -OAuthAccessToken (ConvertTo-SecureString '$(System.AccessToken)' -AsPlainText -Force) -InstanceURL '$(System.CollectionUri)' -RepositoryName 'VS' -ProjectName '$(System.TeamProject)' -CommitIDs $(BaselineBuildCommitIds) -BuildDefinitionIDs 10289 -OutVariableName 'VisualStudio.BaseBuild.ID'
+        - task: PowerShell@2
+          name: SetVisualStudioBaseBuildProductsDropName
+          displayName: Set 'VisualStudio.BaseBuild.ProductsDropName'
+          retryCountOnTaskFailure: 3
+          condition: ne(variables['VisualStudio.BaseBuild.ID'], '')
+          inputs:
+            filePath: $(DartLab.Path)\Scripts\VisualStudio\Build\Get-VisualStudioDropName.ps1
+            arguments: -AccessToken (ConvertTo-SecureString '$(System.AccessToken)' -AsPlainText -Force) -DropNamePrefix 'Products' -AccountURL '$(System.CollectionUri)' -ProjectName '$(System.TeamProject)' -BuildID $(VisualStudio.BaseBuild.ID) -OutVariableName 'VisualStudio.BaseBuild.ProductsDropName'
+        - task: PowerShell@2
+          name: SetBaseProductsDropNameToTarget
+          displayName: Set 'VisualStudio.BaseBuild.ProductsDropName' to drop from target build
+          retryCountOnTaskFailure: 3
+          condition: eq(variables['VisualStudio.BaseBuild.ID'], '')
+          inputs:
+            targetType: 'inline'
+            script: |
+              $buildDrop = "${{parameters.baseBuildDrop}}" -split "/"
+              $dropName = "Products/DevDiv/VS/$($buildDrop[-2])/$($buildDrop[-1])"
```

Comment:
If my understanding is correct, if there is format in the build drop URL then we should tweak these indexes, correct? Do you anticipate any changes to the build drop in the near future?

Reply:
these changes can be breaking changes, so they aren't generally made.